File: src/main/java/com/curiouslad/mekanismextras/util/Registration.java
package com.curiouslad.mekanismextras.util;

import com.curiouslad.mekanismextras.MekanismExtras;
import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class Registration {

    public static final DeferredRegister<Block> BLOCKS
            = DeferredRegister.create(ForgeRegistries.BLOCKS, MekanismExtras.MOD_ID);

    public static final DeferredRegister<Item> ITEMS
            = DeferredRegister.create(ForgeRegistries.ITEMS, MekanismExtras.MOD_ID);
    public static void register()
    {
        IEventBus eventBus = FMLJavaModLoadingContext.get().getModEventBus();
        BLOCKS.register(eventBus);
        ITEMS.register(eventBus);
    }
}

File: src/main/kotlin/com/curiouslad/mekanismextras/util/Registration.kt
package com.curiouslad.mekanismextras.util

import com.curiouslad.mekanismextras.MekanismExtras
import net.minecraftforge.eventbus.api.IEventBus
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.KotlinModLoadingContext.Companion.get

object Registration {
    val BLOCKS = DeferredRegister.create(ForgeRegistries.BLOCKS, MekanismExtras.MOD_ID)
    val ITEMS = DeferredRegister.create(ForgeRegistries.ITEMS, MekanismExtras.MOD_ID)
    fun register() {
        val eventBus: IEventBus = get().getKEventBus()
        BLOCKS.register(eventBus)
        ITEMS.register(eventBus)
    }
}